crates/task-impls/src/consensus/mod.rs

```diff
@@ -435,6 +435,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
                     };
                 }
             },
+            #[cfg(not(feature = "dependency-tasks"))]
```

Comment:
Is this gate redundant? (I thought we didn't spawn regular consensus  with the dependency tasks enabled, though I might have misunderstood)

Reply:
It's actually needed, due to some functions used by both the dependency and non-dependency tasks. But of course, this isn't ideal, so we have a task to separate the shared functions!